### PR TITLE
Push channel management

### PR DESCRIPTION
**Pull request by azlindsay**
Helper methods for create / list / subscribe / unsubscribe.

### CODE REVIEW

File: Sources/Kumulos+PushChannels.swift [hunk 1]
```diff
@@ -0,0 +1,188 @@
+//
+//  Kumulos+PushChannels.swift
```
Comment (cgwyllie) [inline Sources/Kumulos+PushChannels.swift]: If this isn't an extension class then should naming be just the main class name?

File: Sources/Kumulos+PushChannels.swift [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+//  Copyright © 2017 Kumulos. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+
+public typealias PushChannelSubscriptionSuccessBlock = (()->Void)?
+public typealias PushChannelSubscriptionFailureBlock = ((Error?)->Void)?
+
+public class KumulosPushChannelSubscriptionOperation {
```
Comment (cgwyllie) [inline Sources/Kumulos+PushChannels.swift]: Should we call these 'operations' when they don't actually subclass `Operation`? Maybe something like `KumulosPushChannelSubscriptionRequest` or similar, open to suggestions, just think it might be misleading as operations are a framework concept.

File: Sources/Kumulos+PushChannels.swift [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    
+    public init(sdkInstance: Kumulos)     {
+        self.sdkInstance = sdkInstance;
+    }
+    
+    public func listChannels() -> KumulosPushChannelOperation {
+        let operation = KumulosPushChannelOperation()
+        let url =  "\(sdkInstance.basePushUrl)app-installs/\(Kumulos.installId)/channels"
+        
+        _ = sdkInstance.makeNetworkRequest(.get, url: url, parameters: [:])
```
Comment (cgwyllie) [inline Sources/Kumulos+PushChannels.swift]: Can mark the `makeNetworkRequest` as `@discardableResult` and drop the empty assignment?

File: Sources/Kumulos+PushChannels.swift [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        let operation = KumulosPushChannelOperation()
+        let url =  "\(sdkInstance.basePushUrl)channels"
+        
+        var parameters = [
+            "uuid": uuid,
+            "name": name,
+            "showInPortal": showInPortal
+        ] as [String: Any];
+        
+        if (meta.count > 0) {
```
Comment (cgwyllie) [inline Sources/Kumulos+PushChannels.swift]: can we make meta optional (as in nullable) and test it for being a value?

File: Sources/Kumulos+PushChannels.swift [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+        for item in jsonResponse {
+            channels.append(getChannelFromPayload(payload: item))
+        }
+        
+        return channels
+    }
+    
+    private func getChannelFromPayload(payload: [String:AnyObject]) -> PushChannel {
+        let channel = PushChannel()
+        channel.name = payload["name"] as! String
```
Comment (cgwyllie) [inline Sources/Kumulos+PushChannels.swift]: Name can be null, will `as!` explode on this case?

File: Sources/PushChannel.swift [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+//  KumulosSDK
+//
+//  Created by Andy on 07/02/2017.
+//  Copyright © 2017 Kumulos. All rights reserved.
+//
+
+import Foundation
+
+open class PushChannel: NSObject {
+    internal(set) open var name: String = ""
```
Comment (cgwyllie) [inline Sources/PushChannel.swift]: Nullable/optional?

File: Sources/PushChannel.swift [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    internal(set) open var uuid: String = ""
+    internal(set) open var meta: Dictionary<String, AnyObject> = [:]
```
Comment (cgwyllie) [inline Sources/PushChannel.swift]: Nullable/optional instead of defaulting to empty object?

File: Sources/KumulosPushChannels.swift [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+                case .failure(let error):
+                    if let failureBlock = request.failureBlock {
+                        failureBlock?(error)
+                    }
+                }
+        }
+        return request
+    }
+    
+    public func createChannel(uuid: String, name: String, showInPortal: Bool, meta: [String:AnyObject]? = nil) -> KumulosPushChannelRequest
```
Comment (cgwyllie) [inline Sources/KumulosPushChannels.swift]: - Should `name` be nullable?
- Missing the `subscribe` option to subscribe as part of the creation?

File: Sources/KumulosPushChannels.swift [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+    {
+        let request = KumulosPushChannelRequest()
+        let url =  "\(sdkInstance.basePushUrl)channels"
+
```
Comment (cgwyllie) [inline Sources/KumulosPushChannels.swift]: Should add some client-side validation to test if `showInPortal` is set and name is nil/empty, abort? (saves network trip)

File: Sources/KumulosPushChannels.swift [hunk 10]
```diff
@@ -75,11 +79,29 @@ public class KumulosPushChannels {
...
+            - name: Optional descriptive name for the channel
+            - meta: Optional custom meta-data to associate with this push channel
+     */
     public func createChannel(uuid: String, subscribe: Bool, name: String? = nil, meta: [String:AnyObject]? = nil) -> KumulosPushChannelRequest {
         return doCreateChannel(uuid: uuid, subscribe: subscribe, name: name, showInPortal: false, meta: meta)
     }
     
-    
+    /**
+        Create a push channel for subscribing to, it will not be available via the Kumulos portal
```
Comment (cgwyllie) [inline Sources/KumulosPushChannels.swift]: Remove 'it will not be available via the Kumulos portal' ?

File: Sources/KumulosPushChannels.swift [hunk 11]
```diff
@@ -75,11 +79,29 @@ public class KumulosPushChannels {
         return request
     }
     
+    /**
+        Create a push channel for subscribing to, it will not be available via the Kumulos portal
+
```
Comment (cgwyllie) [inline Sources/KumulosPushChannels.swift]: Do we need to explain what happens if 'name' is filled?
